controllers/tasks.js

```diff
@@ -0,0 +1,49 @@
+const Task = require('../models/Task')
+const asyncWrapper = require('../middleware/async')
+const {createCustomError, CustomAPIError} = require('../errors/custom-error')
+
+const getAllTasks = asyncWrapper(async(req,res)=> {
```

Comment:
Well done for creating your first controller and first API functions. Notice that in Node.js indeed the API endpoints are async, but most of the time you should use "await". The concept of "await" is to wait for something syncroniclly, only if you don't have another option. It's less effiencent and hurting the original cause of using aync functions. most of the time, you should create a promises and let the function know what you want her to do, ***after*** the heavy call happend. for example, this is an API enpoint I created in some of my old projects (notice also the documentation, the logging, etc):
```
/**
 * @func    createMachine
 * @desc    Endpoint for creating new machine instance
 * @arg     {body}
 * @argument id - machine's identity number
 * @argument name - machine's name
 * @argument password - machine's passwrod for accessing TakeAway reasorces
 * @argument status - machine's status
 * @argument machineSize - machine's cells number (optional)
 * @argument productsLocations (optional)
 * @argument allowedOrganizationName (optional)
 * @argument allowedOrganizationLevel (optional)
 * @argument description (optional)
 * @returns machine's details in case of success
 */
exports.createMachine = (req, res, next) => {
  // searching the requested identity number
  req.models.Machine.findAll({
    where: {
      identityNumber: req.body.id
    }
  }).then((machines) => {
    if (utilities.noResults(machines)) {
      // creating new machine user in AuthService
      axios.put(USERS_URL, {
        id: req.body.id,
        name: req.body.name,
        password: req.body.password,
        role: { name: MACHINE_ROLE_NAME }
      }, {
        headers: {
          'Authorization': req.headers.authorization
        }
      }).then(() => {
        // creating new instance in machinesdb
        return BuildMachine(req, res, null);
      }).catch(err => {
        logger.error(`Error while trying to create new machine user {${req.body.id}} in AuthService. Original error: ${err}`);
        return res.sendStatus(utilities.statusCodes.ERROR);
      });
    }
    else {
      // if the machine id already exists
      return res.status(utilities.statusCodes.NOT_ACCEPTABLE).send({ message: `The requested machine ${req.body.id} already exists!` });
    }
  }).catch(err => {
    logger.error(`Error while trying to find machine id {${req.body.id}} Original error: ${err}`);
    return res.sendStatus(utilities.statusCodes.ERROR);
  });
}
```